### PR TITLE
chore: adding `IsMiddlewareEnabled` to `ActiveChannel` genesis type

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

closes: #2165 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/27-interchain-accounts/controller/keeper/keeper.go [hunk 1]
```diff
@@ -145,9 +145,10 @@ func (k Keeper) GetAllActiveChannels(ctx sdk.Context) []genesistypes.ActiveChann
 		keySplit := strings.Split(string(iterator.Key()), "/")
 
 		ch := genesistypes.ActiveChannel{
-			ConnectionId: keySplit[2],
-			PortId:       keySplit[1],
-			ChannelId:    string(iterator.Value()),
+			ConnectionId:        keySplit[2],
+			PortId:              keySplit[1],
+			ChannelId:           string(iterator.Value()),
+			IsMiddlewareEnabled: k.IsMiddlewareEnabled(ctx, keySplit[1], string(iterator.Value())),
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/keeper/keeper.go:151]: In the future, I think it'd probably be wise to set these key splits/iterator values to variables for readability purposes. I don't think it needs to change as this is a problem throughout our codebase (keysplit should probably be a parse func)

**Review by colin-axner — APPROVED** [review]
Fantastic! Ultra clean addition

File: modules/apps/27-interchain-accounts/controller/keeper/keeper.go (hunk 1 above)
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/controller/keeper/keeper.go:151]: Agreed! :)

File: proto/ibc/applications/interchain_accounts/genesis/v1/genesis.proto [hunk 2]
```diff
@@ -36,11 +36,13 @@ message HostGenesisState {
   ibc.applications.interchain_accounts.host.v1.Params params = 4 [(gogoproto.nullable) = false];
 }
 
-// ActiveChannel contains a connection ID, port ID and associated active channel ID
+// ActiveChannel contains a connection ID, port ID and associated active channel ID, as well as boolean flag to indicate
```
Comment (crodriguezvega) [inline proto/ibc/applications/interchain_accounts/genesis/v1/genesis.proto:39]:
```suggestion
// ActiveChannel contains a connection ID, port ID and associated active channel ID, as well as a boolean flag to indicate
```

**Review by chatton — APPROVED** [review]
LGTM 👍

**Pull request merged**